plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE ConstraintKinds  #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs            #-}
+{-# LANGUAGE TypeFamilies     #-}
+
+{-|
+Call site inlining machinery. For now there's only one part: inlining of fully applied functions.
+See ADR TODO for motivation. We inline fully applied functions if the cost and size are acceptable.
+See note [Inlining of fully applied functions].
+-}
+
+module PlutusIR.Transform.Inline.CallSiteInline where
+
+{- Note [Inlining of fully applied functions]
+
+We inline the call site when a function is fully applied AND the cost and size are acceptable.
```

Comment:
cost and size of _what_? The function? the body of the function? be specific!

Reply:
clarified in the bottom.